src/index.tsx

```diff
@@ -25,23 +25,40 @@ import * as serviceWorker from "./serviceWorker";
 import { store } from "./reducers/store";
 import { Provider } from "react-redux";
 
+import { useDispatch } from "react-redux";
+import { syncWithLocalStorage } from "./actions/localStorageActions";
+
 import { DndProvider } from "react-dnd";
 import { HTML5Backend } from "react-dnd-html5-backend";
 
 const AppWithPersistence = () => {
   store.subscribe(() => {
     const reduxState = store.getState();
-    const draftMessages = {
+    const localStorageState = {
       authors: reduxState.authors,
       messages: reduxState.messages,
       points: reduxState.points,
       draftMessages: reduxState.draftMessages,
       draftPoints: reduxState.draftPoints,
-      semanticScreen: reduxState.semanticScreen,
       userIdentities: reduxState.userIdentities,
     };
 
-    localStorage.setItem("draftMessages", JSON.stringify(draftMessages));
+    localStorage.setItem(
+      "localStorageState",
+      JSON.stringify(localStorageState)
+    );
+  });
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    window.onstorage = () => {
```

Comment:
What is this event for?

Reply:
Ah, I see, it's to detect when another tab updates the storage?

Reply:
That's correct

Reply:
Thank you for looking it over!